### ISSUE TITLE
Port sanitizer runners to GitLab CI

### ISSUE THREAD

**Issue by j-stephan** [alpaka-group/alpaka#2097]
Our current sanitizer runners are split between GitHub and GitLab:

* UBsan runs on GitHub
* TSan runs on GitHub
* ASan runs on GitLab

They should all be migrated to GitLab.

**Comment by psychocoderHPC** [comment]
As I know @SimeonEhrig is working on bringing CPU CI tests to GitLab, I assume the analysis test will be moved too.

**Comment by SimeonEhrig** [comment]
That's true. We already discussed, how we can simply generate sanitizer jobs from the common CPU jobs. At the moment, I'm work on GCC jobs.

**Comment by bernhardmgruber** [comment]
Great, I will leave this to @SimeonEhrig then!
